scalac-scoverage-plugin/src/main/scala/scoverage/plugin.scala

```diff
@@ -35,16 +34,31 @@ class ScoveragePlugin(val global: Global) extends Plugin {
         options.dataDir = opt.substring("dataDir:".length)
       } else if (opt.startsWith("extraAfterPhase:") || opt.startsWith("extraBeforePhase:")) {
         // skip here, these flags are processed elsewhere
-      } else {
+      } else if (opt.startsWith("writeToClasspath:")) {
+        try {
+          options.writeToClasspath = opt.substring("writeToClasspath:".length).toBoolean
+        } catch {
+          case _: IllegalArgumentException => throw new RuntimeException("writeToClasspath can only be a boolean value: true or false")
+          case e: Exception => throw e
+        }
+      } else{
         error("Unknown option: " + opt)
       }
     }
-    if (!opts.exists(_.startsWith("dataDir:")))
-      throw new RuntimeException("Cannot invoke plugin without specifying <dataDir>")
+
+    if (opts.exists{p: String => p.contains("writeToClasspath") && p.contains("false")} || (!opts.exists(_.startsWith("writeToClasspath")))) {
```

Comment:
It would be good to extract this entire condition and give it a name: for example: ```
val isUsingWriteToClasspath: Boolean = ???
if (!isUsingWriteToClasspath) { ??? }
```

Reply:
with the latest change (below), we always need the dataDir to be set.